src/types.rs

```diff
@@ -418,6 +419,7 @@ impl<'a> NewOrder<'a> {
         Self {
             identifiers,
             replaces: None,
+            profile: None,
```

Comment:
I think similar to ARI we should also gain a `profile: Option<String>` field on the `OrderState`.

That would let `NewOrder`'s that _don't_ specify a `profile` learn what "default" profile the server assigned their order:

>  If the server is advertizing profiles and receives a newOrder request which does not identify a specific profile, it is RECOMMENDED that the server select a profile and associate it with the new Order object.

Relatedly, the profiles spec doesn't say the server MUST reflect the matching `profile` for new order requests that _do_ specify a profile. That's probably good feedback to give to the author. I think ideally we would enforce it matches like for ARI.

Reply:
The spec isn't specific at all about whether the order state will actually reflect back the selected profile, I think?

Reply:
Yeah, I think there's language missing to make that crystal clear. I'm partly inferring it based on the [registry update](https://www.ietf.org/archive/id/draft-aaron-acme-profiles-00.html#section-6.2) and the accompanying Pebble logic:

* Here's where it [assigns a default if one isn't provided](https://github.com/letsencrypt/pebble/blob/dc9bf9d200b49f5b587258ec53948f17417a7579/wfe/wfe.go#L1728-L1744)
* And here's where the profile name (provided _or_ the default choice) is [populated in the order state](https://github.com/letsencrypt/pebble/blob/dc9bf9d200b49f5b587258ec53948f17417a7579/wfe/wfe.go#L1776)

Reply:
It's also similar [in the Boulder order JSON](https://github.com/letsencrypt/boulder/blob/53c35ac669b212ebcda95c7956d5a1fb2f0d4b1e/wfe2/wfe.go#L1959)

Reply:
Also, I couldn't think of a great way to test this.

Reply:
Yeah I think the Pebble test code makes it annoying to get at the order state while also using all the shared logic for completing the order. Seems fine to not test it explicitly for now. I think the spec should be updated to be clearer and then we can check it in the `lib.rs` code.

Reply:
> The spec isn't specific at all about whether the order state will actually reflect back the selected profile, I think?

@djc It looks like this [got addressed](https://author-tools.ietf.org/iddiff?url2=draft-aaron-acme-profiles-01) in -01 :tada: 

![diff](https://github.com/user-attachments/assets/8fe1e0f3-6031-42ee-aa74-372d5f57015f)